sdk/samples/iot/paho_iot_hub_pnp_component_sample.c

```diff
@@ -882,7 +879,15 @@ static void process_twin_message(
           exit(rc);
         }
 
-        rc = append_simple_json_token(&jw, &property_value.token);
+        rc = az_json_reader_next_token(&property_name_and_value);
+        if (az_result_failed(rc))
+        {
+          IOT_SAMPLE_LOG_ERROR(
+              "Could not advance to the property value: az_result return code 0x%08x.", rc);
+          exit(rc);
+        }
+
+        rc = append_simple_json_token(&jw, &property_name_and_value.token);
```

Comment:
Same as my comment on the other PR, I would use the IOT_SAMPLE_EXIT_IF_AZ_FAILED in these spots since there are many repetitive log and exits:

```
rc = az_json_reader_next_token(&property_name_and_value);
IOT_SAMPLE_EXIT_IF_AZ_FAILED(rc, "Could not advance to the property value");
```

Please note, with these the colon and return code part of the comment are appended in the macro, so no need to include here.

Reply:
noted tackling in another PR since there are other places where this could be improved